### PR TITLE
fix(traffic_light_arbiter): publish traffic light info even when a map without traffic lights is subscribed

### PR DESCRIPTION
**Pull request by tkimura4**
## Description
 publish traffic light info even when a map without traffic lights is subscribed

## Tests performed
I tested as below.

- launch traffic_light_arbiter
```
$ros2 launch traffic_light_arbiter traffic_light_arbiter.launch.xml 
```

- publish map without traffic lights

- publish /external/traffic_light_recognition/traffic_signals 
```
$ros2 topic pub /external/traffic_light_recognition/traffic_signals autoware_perception_msgs/msg/TrafficSignalArray {}
```

- confirm that traffic_signals messages are published by traffic_light_arbiter
```
$ ros2 topic echo /perception/traffic_light_arbiter/traffic_signals
```

## Effects on system behavior
Even when you use a map without traffic light, traffic_light_arbiter will publish traffic light info.
( When you use a map with traffic light, this PR has no effects on system behavior. )

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: perception/traffic_light_arbiter/include/traffic_light_arbiter/traffic_light_arbiter.hpp [hunk 1]
```diff
@@ -45,7 +46,7 @@ class TrafficLightArbiter : public rclcpp::Node
   void onExternalMsg(const TrafficSignalArray::ConstSharedPtr msg);
   void arbitrateAndPublish(const builtin_interfaces::msg::Time & stamp);
 
-  std::unordered_set<lanelet::Id> map_regulatory_elements_set_;
+  std::shared_ptr<std::unordered_set<lanelet::Id>> map_regulatory_elements_set_;
```
Comment (knzo25) [inline perception/traffic_light_arbiter/include/traffic_light_arbiter/traffic_light_arbiter.hpp:49]:
```suggestion
  std::unique_ptr<std::unordered_set<lanelet::Id>> map_regulatory_elements_set_;
```

File: perception/traffic_light_arbiter/src/traffic_light_arbiter.cpp [hunk 2]
```diff
@@ -72,9 +72,10 @@ void TrafficLightArbiter::onMap(const LaneletMapBin::ConstSharedPtr msg)
   lanelet::utils::conversion::fromBinMsg(*msg, map);
 
   const auto signals = lanelet::filter_traffic_signals(map);
-  map_regulatory_elements_set_.clear();
+  map_regulatory_elements_set_ = std::make_shared<std::unordered_set<lanelet::Id>>();
```
Comment (knzo25) [inline perception/traffic_light_arbiter/src/traffic_light_arbiter.cpp:75]:
```suggestion
  map_regulatory_elements_set_ = std::make_unique<std::unordered_set<lanelet::Id>>();
```

**Review by knzo25 — CHANGES_REQUESTED** [review]
Thanks for the PR.
Tested it using the data from `2023-04-17-17-44-15`.
Only one nit: shared_ptr -> unique_ptr as the data is not being shared

**Review by knzo25 — APPROVED** [review]
LGTM！

**Comment by tkimura4** [comment]
@knzo25 
Thank you for your review!!!

**Pull request merged**